src/liteclient.cc

```diff
@@ -836,3 +827,22 @@ bool LiteClient::isRegistered(const KeyManager& key_manager) {
   }
   return false;
 }
+
+LiteClient::Type LiteClient::getClientType(const KeyManager& key_manager) {
+  static const std::string prod_bc_value{"production"};
+
+  Type res{Type::Undefined};
+  try {
+    const auto bc{key_manager.getBC()};
+    if (bc == prod_bc_value) {
+      res = Type::Prod;
+      LOG_DEBUG << "Found production business category in a device certificate: " << bc;
+    } else {
+      res = Type::Dev;
+      LOG_DEBUG << "Missing or non-production business category found in a device certificate: " << bc;
+    }
+  } catch (const std::exception& exc) {
+    LOG_WARNING << "A device certificate is not found or failed to parse it: " << exc.what();
```

Comment:
If there is no client certificate or we failed to parse it - how can the device connect to our APIs?
Shouldn't we just fail hard in this case?

Reply:
In the case of an offline update connection to the API is not required.

Reply:
Just double checking.
Do you mean that having a device certificate is not required for offline device too?

That makes sense to me.

Reply:
>Do you mean that having a device certificate is not required for offline device too?

Yes.